### PR TITLE
DBZ-6838: Blog post for addition of Custom properties step in DBZ UI

### PR DESCRIPTION
**Pull request by indraraj**

### CODE REVIEW

File: _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
++++<!-- more -->+++
+
+== Need for custom properties step
+
+Debezium is very flexible - each connector can be configured and fine-tuned in a variety of ways. Debezium UI already expose an extensive list of properties that are segregated under different steps for configuration. But even with this extensive list of exposed properties in UI there are few properties that are not know prior and are need to configure while connector setup.
+
+*For example*
+
+Let’s take an example of oracle connector.
```
Comment (indraraj) [inline _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc]: @Naros @rk3rn3r I am looking for example properties we may provide as additional properties here for a connector. And some bit of reasoning like why we don't have prior knowledge and could not be exposed via UI

Comment (rk3rn3r) [reply _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc]: I think all `producer.*` and `consumer.*` properties are additional options, as well as all pass-through options for the database drivers starting with `database.` and that are not part of the UI properties / the documented schema.

Comment (Naros) [reply _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc]: Hi @rk3rn3r and @indraraj, just one reminder that we changed the `database.` prefix to `driver.` prefix for passthru driver properties in Debezium 2.0.  The only properties that begin with `database.` should generally focus only on values used by Debezium proper.

Oracle properties can be found at:
https://docs.oracle.com/en/database/oracle/oracle-database/21/jajdb/oracle/jdbc/OracleConnection.html

Some example driver properties that might be useful for Oracle `oracle.net.keepAlive` which is useful when deploying on AWS with GLB enabled, `oracle.net.disableOob` if the OOB isn't disabled at the TNS listener level, as well as various timeouts managed by the driver.

**Comment by indraraj** [comment]
@rk3rn3r @Naros FYI ready for review

File: _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
+---
+layout: post
+title:  Enable pass-through of additional properties in Debezium UI
+date:   2023-08-21
+tags: [ community, debezium-ui ]
+author: indrashukla
+---
+The Debezium UI team continues to add support for more features, allowing users to configure connectors more easily. In this article, we'll describe and demonstrate how to provide the additional properties for configuration that are not specified in the UI.
```
Comment (Naros) [inline _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc]:
```suggestion
The Debezium UI team continues to add support for more features, allowing users to configure connectors more easily. In this article, we'll describe and demonstrate how to provide the additional properties for configuration that the UI does not expose by default.
```

File: _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+author: indrashukla
+---
+The Debezium UI team continues to add support for more features, allowing users to configure connectors more easily. In this article, we'll describe and demonstrate how to provide the additional properties for configuration that are not specified in the UI. 
+Read further for more information!
+
++++<!-- more -->+++
+
+== Need for custom properties step
+
+Debezium is very flexible - each connector can be configured and fine-tuned in a variety of ways. Debezium UI already exposes an extensive list of properties that are segregated under different steps for configuration. But even with this extensive list of exposed properties in the UI, there are a few properties that are not known prior and may be needed to be configured while setting up a connector. For example, `producer.*` and `consumer.*` properties are additional options, similarly `driver.*` prefix for the pass-through database drivers properties are also not exposed via UI and if needed, these are to be configured as additional properties.
```
Comment (Naros) [inline _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc]:
```suggestion
Debezium is very flexible - each connector can be configured and fine-tuned in various ways. Debezium UI already exposes an extensive list of properties that are segregated under different steps for configuration. However, despite the UI exposing a wide range of properties, there are situations where additional custom properties may be necessary beyond those exposed out of the box. For example, `producer.*`, `consumer.*`, and even `driver.*` properties are pass-through configuration options for configuring parts of the Kafka Connect pipeline or the JDBC driver, respectively.
```

File: _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+*Example*
+ 
+Let's take an example of mySql connector, debezium relies on a kafka producer to write schema changes to the database history topic. You define the configuration for the kafka producer by assigning the values to a set of pass-through configuration properties that begin with the `schema.history.internal.producer.*` prefix.
```
Comment (Naros) [inline _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc]:
```suggestion
Debezium must configure the Kafka producer settings to write schema change events to the schema history topic for connectors that utilize a schema history topic, like MySQL. The connector configuration includes a subset of options that act as pass-through properties that begin with the `schema.history.internal.producer.*` prefix.
```

File: _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+Debezium strips the prefix from the properties name before it passes the properties to the kafka client.
```
Comment (Naros) [inline _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc]:
```suggestion
Debezium strips the prefix from the property names before passing the properties to the Kafka client.
```

File: _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+== Custom properties step
+
+The Debezium UI now allows you to configure additional properties for connectors that are not exposed in the UI. For this we have created a new additional step called *Custom Properties* in the configuration wizard.
```
Comment (Naros) [inline _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc]:
```suggestion
The Debezium UI allows you to configure additional properties for connectors not exposed in the UI. For this, we have created a new extra step called *Custom Properties* in the configuration wizard. 
```

File: _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+The Debezium UI now allows you to configure additional properties for connectors that are not exposed in the UI. For this we have created a new additional step called *Custom Properties* in the configuration wizard. 
+
+[.centered-image.responsive-image]
+====
+++++
+<img src="/assets/images/2023-08-20-enable-passthru-ui/CustomStep2.png" class="responsive-image">
+++++
+====
+
+*Custom properties* step allow you to provide the additional properties by entering `key` and `value` pair. Once you enter the `key` and `value` pairs of the additional properties, click on the *Apply* button.
```
Comment (Naros) [inline _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc]:
```suggestion
The *custom properties* step allows you to provide the additional properties by entering a key-value pair. Once you enter the additional configuration properties `key` and `value`, click the *Apply* button.
```

File: _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+[.centered-image.responsive-image]
+====
+++++
+<img src="/assets/images/2023-08-20-enable-passthru-ui/configureEmptylist.png" class="responsive-image">
+++++
+====
+
+On the *Apply* button click UI will validate the user entries and provide feedback, if there are any problems. If there are no problems, the UI will show a green check icon on the right side of all the successfully added additional properties. You can also remove any added property by clicking on the *Remove* button on the right side of the property.
```
Comment (Naros) [inline _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc]:
```suggestion
When clicking the *Apply* button, the UI validates the user entries and provides feedback if there are any problems. If there are no problems, the UI will show a green check icon on the right side of all the successfully added additional properties. You can also remove any added property by clicking on the *Remove* button on the right side of the property.
```

File: _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+On the left hand side, you can see the list of already configured properties from previous wizard step and the newly added custom properties in this step are show highlighted in the *blue* color.
```
Comment (Naros) [inline _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc]:
```suggestion
On the left-hand side, you can see the list of already configured properties from the previous wizard step, and the newly added custom properties in this step are highlighted in *blue* color.
```

File: _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+[.centered-image.responsive-image]
+====
+++++
+<img src="/assets/images/2023-08-20-enable-passthru-ui/CustomProp.png" class="responsive-image">
+++++
+====
+
+In the screenshot above user added `schema.history.internal.producer.*` related additional properties .
```
Comment (Naros) [inline _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc]:
```suggestion
In the screenshot above, the user added `schema.history.internal.producer.*` related additional properties.
```

File: _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+
+== Self-contained example
+
+You can try out configuring connector with some custom properties (and more) with our self-contained example https://github.com/debezium/debezium-examples/tree/main/ui-demo[UI demo] - which is included under https://github.com/debezium/debezium-examples[debezium-examples] on Github.  The UI demo includes a Docker Compose file which brings up several sources with data as well as the UI. Please refer to the https://github.com/debezium/debezium-examples/tree/main/ui-demo[README file] for more details on running the Debezium UI demo.
```
Comment (Naros) [inline _posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc]:
```suggestion
You can try out configuring the connector with some custom properties (and more) with our self-contained example https://github.com/debezium/debezium-examples/tree/main/ui-demo[UI demo] - which is included under https://github.com/debezium/debezium-examples[debezium-examples] on GitHub.  The UI demo includes a Docker Compose file, which brings up several sources with data and the UI. Please refer to the https://github.com/debezium/debezium-examples/tree/main/ui-demo[README file] for more details on running the Debezium UI demo.
```

**Review by Naros — COMMENTED** [review]
Hi @indraraj, I've made some suggestions below.  

Also, please be sure to put each sentence on a line by itself in the future. It makes later reviews, particularly on documentation, easier and makes conflicts less problematic when doing merges.

Lastly, could you also update the publication date and the filename prefix to `2023-09-07`  so we can merge it and it correctly shows chronologically in the blog?

**Comment by indraraj** [comment]
@Naros thanks for the review, I have updated the date. Let me know if it's ready to merge.

**Comment by Naros** [comment]
Thanks @indraraj !

**Pull request merged**